src/SFML/Window/WindowBase.cpp

```diff
@@ -146,22 +168,22 @@ bool WindowBase::isOpen() const
 
 
 ////////////////////////////////////////////////////////////
-Event WindowBase::pollEvent()
+std::optional<Event> WindowBase::pollEvent()
 {
-    Event event;
-    if (m_impl && (event = m_impl->pollEvent()))
-        filterEvent(event);
-    return event;
+    return WindowsBaseImpl::pollOrWaitEventImpl(
+        m_impl.get(),
+        [this] { return m_impl->pollEvent(); },
+        [this](sf::Event& e) { filterEvent(e); });
```

Comment:
I personally think this is overabstraction, but won't demand its removal. I get how it works now but don't love how long it took to understand how it works compared to the naive alternative of having two functions with nearly identical implementations. I don't wish to see more of this pattern going forward.

Reply:
Fair enough, the repetition is not that bad here, so I will remove the template function. I think this pattern is still valuable when there's a larger amount of repetition to ensure things are kept in sync and there are no copy-paste mistakes, hopefully you'll agree with that for future cases where it's more valuable.